linux/centos/7/foxpass_setup.py

```diff
@@ -138,38 +145,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
 
 
 def augment_sshd_config():
-    if not file_contains('/etc/ssh/sshd_config', '^AuthorizedKeysCommand'):
+    if not file_contains('/etc/ssh/sshd_config', r'^AuthorizedKeysCommand '):
         with open('/etc/ssh/sshd_config', "a") as w:
             w.write("\n")
             w.write("AuthorizedKeysCommand\t\t/usr/local/sbin/foxpass_ssh_keys.sh\n")
             w.write("AuthorizedKeysCommandUser\troot\n")
 
 
 # give "wheel" and chosen sudoers groups sudo permissions without password
-def fix_sudo(sudoers):
-    os.system("sed -i 's/^# %wheel\tALL=(ALL)\tNOPASSWD: ALL/%wheel\tALL=(ALL)\tNOPASSWD:ALL/' /etc/sudoers")
-    if not file_contains('/etc/sudoers', '^#includedir'):
+def fix_sudo(sudoers, require_sudoers_pw):
+    if not file_contains('/etc/sudoers', r'^#includedir /etc/sudoers.d'):
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
+    if not require_sudoers_pw:
+        os.system("sed -i 's/^%wheel\tALL=(ALL:ALL) ALL/%wheel ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers")
```

Comment:
this is different than the previous `sed` line. on purpose?

Reply:
Copy Pasta error
Got through testing because of the foxpass_sudoers group (need to update that)